custom_components/extron/media_player.py

```diff
@@ -109,10 +121,13 @@ def source(self):
 
     @property
     def source_list(self):
-        return self._source_list
+        return list(self._source_bidict.values())
+
+    def make_source_bidict(self) -> bidict:
+        return make_source_bidict(5, self._input_names)
```

Comment:
You have the same function name for both is is a bit confusing. I would rename the other method, for example
```suggestion
    def make_source_bidict(self) -> bidict:
        return create_source_bidict(5, self._input_names)
```

Reply:
Renamed this at the least minute, didn't notice the clash